test/alternator/test_tracing.py

```diff
@@ -266,8 +266,10 @@ def test_slow_query_log(with_slow_query_logging, test_table_s, dynamodb):
     delay = 0.2
     while delay < 30:
         results = full_scan(slow_query_table, ConsistentRead=False)
-        put_item_found = any("PutItem" in result['parameters'] and p in result['parameters'] for result in results)
-        delete_item_found = any("DeleteItem" in result['parameters'] and p in result['parameters'] for result in results)
+        put_item_found = any("PutItem" in result['parameters'] and p in result['parameters']
+                and result['username'] == "alternator" for result in results)
```

Comment:
I would prefer to have a separate test function that just checks the 'username' - not merged with an existing test function. My thinking is that maybe someone would like to run this test suite against an already-running Alternator, and maybe it doesn't have this "alternator" user. A person running such a setup would not be surprised if all the tests in test_authorization.py fail, and if one test function called test_tracing_username (or something) fails - but it would be surprising that "test_slow_query_log" fails just because of the authorization thing inside it.

HOWEVER, all that being said, I wonder if there's a way to do this new separate test quickly :-( I remember there was an annoying 1-second delay in flushing the trace tables. I wonder if it's a configurable parameter, that we can change in test/alternator/run so it will be 0.1 second instead of 1 second (or whatever).

Reply:
We already have a `test_authorization.py` suite that makes an assumption that the server has auth set up for username: `alternator` and password: `secret_pass`. If somebody wants to run it locally (as I personally often do), these tests can either be skipped with ` pytest --ignore=<PATH>` or the credentials can be simply set up locally.

Adding this as a new test case will be a rather ugly copy-paste, which is why I piggy-backed on an existing test. Thing is - we cannot really assume that there are no other tracing events from previous runs (if the server is standalone), so I cannot just check that "all traces contain a valid username". Instead, I'd have to send very specific requests and then verify if they also contain a valid username, which pretty much means that I'd copy the whole slow logging test case and add a single line of username verification - so I ultimately decided it's better to just add this single line to the existing test case. It reduces execution time and code duplication at the cost of not allowing users to run this single slow query logging test case without auth properly configured, which I think is a fair price.

Reply:
Perhaps. What I'm worried about is that you are already used to ignoring the test_autorization.py on your tests, but now you'll need to remember to ignore another one - one which doesn't have any obvious relationship to authorization, and you only notice this relation if you inspect the code.
But I guess we can always fix this later too, if it bothers us. I'll commit the next version you push.